15/umbraco-cms/fundamentals/setup/server-setup/running-umbraco-in-docker.md

```diff
@@ -0,0 +1,95 @@
+# Running Umbraco in Docker
+
+Exactly how you choose to compose your Dockerfile depends on your needs, and your project. This section is not intended as a guide, 
+but as a general overview of what to be aware of when hosting in Docker.
+
+## What is Docker
+
+Docker is a platform for developing, shipping, and running applications in containers. There exist multiple services for hosting these containers, 
+for more information, [refer to the official Docker documentation](https://docs.docker.com/)
+
+## The Docker file system
+
+By default, all files created inside a container are written to an ephemeral writable container layer. 
+This means that the files don't persist when the container is removed, and it's challenging to get files out of the container. Additionally, this writable layer is not suitable for performance-critical data processing.
+This has implications when running Umbraco in Docker. For more information, refer to the [Docker documentation on storage](https://docs.docker.com/engine/storage/).
+
+### General file system consideration 
+
+In general, when working with files and Docker you work in a "push" fashion with the read-only layers. WWhen you build, you take all your files and "push" them into the read-only layer.
+This means that you should avoid making files on the fly, and instead rely on building your image. You should not create or edit template files on the fly, the same goes for script and style files. 
+
+Similarly, you shouldn't use InMemory modelsbuilder, since that also relies on creating files on the disk. While this is not a hard requirement, it doesn't provide any value if not live editing your site. Instead, use source code in development, and none in production, as [described when using runtime modes](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/runtime-modes).
```

Comment:
```suggestion
Similarly, you shouldn't use InMemory modelsbuilder, since that also relies on creating files on the disk. While this is not a hard requirement, it doesn't provide any value if not live editing your site.

Instead, configure models builder to use use "source code" mode in development, and "none" in production, as [described when using runtime modes](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/runtime-modes).
```